packages/web-components/src/components/ic-menu/ic-menu.spec.tsx

```diff
@@ -0,0 +1,963 @@
+    const page = await createMenu();
+
+    page.root.open = false;
+    expect(page.root.open).toBe(false);
+  });
+  it("tests ungroupedOptionsSet emitter", async () => {
+    const page = await createMenu();
+
+    const eventSpy = jest.fn();
+
+    page.root.addEventListener("ungroupedOptionsSet", eventSpy);
+
+    page.root.options = [];
+
+    await page.waitForChanges();
+
+    expect(eventSpy).toHaveBeenCalledWith(
+      expect.objectContaining({
+        detail: expect.objectContaining({
+          options: [],
+        }),
+      })
+    );
+  });
+  it("tests menuOptionSelect emitter", async () => {
+    const page = await createMenu();
+
+    const eventSpy = jest.fn();
+
+    page.root.addEventListener("menuOptionSelect", eventSpy);
+
+    await page.waitForChanges();
+
+    page.rootInstance.setInputValue(2);
+
+    await page.waitForChanges();
+
+    expect(eventSpy).toHaveBeenCalledWith(
+      expect.objectContaining({
+        detail: expect.objectContaining({
+          value: "flatwhite",
+        }),
+      })
+    );
+  });
+  it("tests menuStateChange emitter", async () => {
+    const page = await createMenu();
+
+    const eventSpy = jest.fn();
+
+    page.root.addEventListener("menuStateChange", eventSpy);
+
+    await page.waitForChanges();
+
+    const click = new MouseEvent("click", {
+      bubbles: true,
+      cancelable: true,
+    });
+
+    page.rootInstance.arrowBehaviour(click);
+
+    await page.waitForChanges();
+
+    expect(eventSpy).toHaveBeenCalledWith(
+      expect.objectContaining({
+        detail: expect.objectContaining({
+          open: true,
+        }),
+      })
+    );
+  });
+  it("tests menuOptionId emitter", async () => {
+    const page = await createMenu();
+
+    const eventSpy = jest.fn();
+
+    page.root.addEventListener("menuOptionId", eventSpy);
+
+    await page.waitForChanges();
+
+    const keyboardEvent = {
+      key: "ArrowDown",
+      preventDefault: (): void => null,
+    };
+
+    page.rootInstance.manSetInputValueKeyboardOpen(keyboardEvent);
+
+    await page.waitForChanges();
+
+    expect(eventSpy).toHaveBeenCalledWith(
+      expect.objectContaining({
+        detail: expect.objectContaining({
+          optionId: "menu-id-espresso",
+        }),
+      })
+    );
+  });
+  it("tests menuKeyPress emitter", async () => {
+    const page = await createMenu();
+
+    const eventSpy = jest.fn();
+
+    page.root.addEventListener("menuKeyPress", eventSpy);
+
+    await page.waitForChanges();
+
+    const keyboardEvent = {
+      key: "Home",
+      preventDefault: (): void => null,
+    };
+
+    page.rootInstance.autoSetValueOnMenuKeyDown(keyboardEvent);
+
+    await page.waitForChanges();
+
+    expect(eventSpy).toHaveBeenCalledWith(
+      expect.objectContaining({
+        detail: expect.objectContaining({
+          isNavKey: true,
+          key: "Home",
+        }),
+      })
+    );
+  });
+});
+describe("ic-menu in ic-select", () => {
```

Comment:
think it would be fine to leave these sorts of tests in ic-select.spec, but make sure we can get as much code coverage as possible without needing to rely on that. That way we know we can remove from the ic-select in future if\when it is refactored, but have confidence that we have coverage for ic-menu independently.